src/views/reports/base/RightAction.vue

```diff
@@ -56,17 +56,16 @@ export default {
       }
       const days = this.getDaysParam()
       const exportUrl = `/core/reports/export-pdf/?chart=${this.name}&days=${days}`
-      download(exportUrl)
       this.$message.success(this.$t('Export') + '...')
+      download(exportUrl)
     },
     emailReport() {
       if (!this.checkName()) {
         return
       }
       const days = this.getDaysParam()
-      this.$axios.post(`/core/reports/send-mail/?chart=${this.name}&days=${days}`,).then((data) => {
-        this.$message.success(this.$t('EMailReport') + '...')
-      }).catch(error => {
+      this.$message.success(this.$t('EMailReport') + '...')
+      this.$axios.post(`/core/reports/send-mail/?chart=${this.name}&days=${days}`,).catch(error => {
         this.$message.error(this.$t('Failed') + ': ' + error.message)
       })
     },
```

Comment:
There were no issues detected in the provided code samples.

In terms of recommendations for optimization, as we're only considering differences in 2021 which may have been influenced by older technologies or practices that might not apply to today's requirements:

- Use ES Module syntax instead of CommonJS.
- Ensure all modules and scripts are included at the top level (not after the window object).
- Consider using promises/async/await more consistently.

Note: This is based on historical standards and could vary slightly depending on newer coding best practices adopted later in development cycles.